rust/parquet/src/arrow/levels.rs

```diff
@@ -0,0 +1,692 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Contains the logic for computing definition and repetition levels
+
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub(crate) struct LevelInfo {
+    /// Array's definition levels
+    pub definition: Vec<i16>,
+    /// Array's optional repetition levels
+    pub repetition: Option<Vec<i16>>,
+    /// Definition mask, to indicate null ListArray slots that should be skipped
+    pub definition_mask: Vec<(bool, i16)>,
```

Comment:
I don't understand how `definition` and `definition_mask` differ...

Reply:
I'll have to write this up, as it becomes relevant when dealing with lists.

Reply:
I added some details, but if it's still unclear, I'll be able to better demostrate it with nested lists :)